### PR TITLE
Fix signal teardown outside main thread

### LINKED ISSUE #11118: ValueError('signal only works in main thread') messages back for wand sweeps in v1.5.6
## 🐛 Bug

An issue raised in [#10336](https://github.com/PyTorchLightning/pytorch-lightning/issues/10336) is back in a lesser form in `pl` v1.5.6.

After upgrading to v1.5.6, wandb hyperparameter sweeps performed in Colab notebooks raise a`ValueError`:

```python
ValueError('signal only works in main thread')
```
  Unlike in [#10336](https://github.com/PyTorchLightning/pytorch-lightning/issues/10336), the error does not terminate the sweep, which still seems to sync up with `wandb` without issue, as far as I can tell.

### To Reproduce

Reproduced at the end of this minimally modified BoringModel Colab notebook: [https://colab.research.google.com/drive/1QheDfu4G5QEUSnHWpvK7UZzdHysRQq4h?usp=sharing](https://colab.research.google.com/drive/1QheDfu4G5QEUSnHWpvK7UZzdHysRQq4h?usp=sharing)

### Expected behavior

`wandb` sweeps should complete without the cited `ValueError`. This was previously fixed in `pl` v1.5.3 and v1.5.4

### Environment

* CUDA:
	- GPU:
		- A100-SXM4-40GB
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0+cu111
	- pytorch-lightning: 1.5.6
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.12
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021
You can also fill out the list below manually.
-->

- PyTorch Lightning Version (e.g., 1.3.0):
- PyTorch Version (e.g., 1.8)
- Python version:
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #11118

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Nice catch !

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
